pkgs/build-support/closure-info.nix

```diff
@@ -21,14 +21,22 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = [ coreutils jq ];
 
+  empty = rootPaths == [];
+
   buildCommand =
     ''
       out=''${outputs[out]}
 
       mkdir $out
 
-      jq -r ".closure | map(.narSize) | add" < "$NIX_ATTRS_JSON_FILE" > $out/total-nar-size
-      jq -r '.closure | map([.path, .narHash, .narSize, "", (.references | length)] + .references) | add | map("\(.)\n") | add' < "$NIX_ATTRS_JSON_FILE" | head -n -1 > $out/registration
-      jq -r '.closure[].path' < "$NIX_ATTRS_JSON_FILE" > $out/store-paths
+      if [[ -n "$empty" ]]; then
+        echo 0 > $out/total-nar-size
```

Comment:
To be consistent with the version from before this should actually be `echo null` I'm afraid:

```
$  nix --version                                                                       
nix (Nix) 2.17.1
$ git describe       
23.11-pre-56676-gf126ff2c29d7
$ cat $(nix-build -E 'with import ./. {}; closureInfo { rootPaths = []; }')/total-nar-size
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/y6d6xdymhrcq400ydfql65ka6hxay7d5-closure-info
copying path '/nix/store/y6d6xdymhrcq400ydfql65ka6hxay7d5-closure-info' from 'https://cache.nixos.org'...
null
```

Reply:
I don't like this and we should ignore that particular behaviour imho. :D

Reply:
So, can we get an agreement on some change unblocking this?  We certainly can't merge staging-next now, as the issue would block both NixOS channels.

Reply:
Sorry I didn't mean to block this. It would've been nice to have the commit message fixed, but fixing staging-next is more important.